src/frontend/src/binder/expr/function/mod.rs

```diff
@@ -483,30 +488,15 @@ impl Binder {
 
         // The actual inline logic for sql udf
         // Note that we will always create new udf context for each sql udf
-        let Ok(context) = UdfContext::create_udf_context(&args, &func) else {
-            return Err(ErrorCode::InvalidInputSyntax(
-                "failed to create the `udf_context`, please recheck your function definition and syntax".to_string()
-            )
-            .into());
-        };
-
         let mut udf_context = HashMap::new();
-        for (c, e) in context {
-            // Note that we need to bind the args before actual delve in the function body
-            // This will update the context in the subsequent inner calling function
-            // e.g.,
-            // - create function print(INT) returns int language sql as 'select $1';
-            // - create function print_add_one(INT) returns int language sql as 'select print($1 + 1)';
-            // - select print_add_one(1); # The result should be 2 instead of 1.
-            // Without the pre-binding here, the ($1 + 1) will not be correctly populated,
-            // causing the result to always be 1.
-            let Ok(e) = self.bind_expr(e) else {
-                return Err(ErrorCode::BindError(
-                    "failed to bind the argument, please recheck the syntax".to_string(),
-                )
-                .into());
-            };
-            udf_context.insert(c, e);
+        for (i, arg) in args.into_iter().enumerate() {
+            if func.arg_names[i].is_empty() {
+                udf_context.insert(format!("${}", i + 1), arg);
+            } else {
+                // The index mapping here is accurate
+                // So that we could directly use the index
+                udf_context.insert(func.arg_names[i].clone(), arg);
+            }
```

Comment:
Actually, I don't quite get how this context works. Could you please elaborate it more?

Reply:
The context contains a mapping of `argument name -> argument expression`. If an argument has no name, `$n` will be used as its name. I just updated the comment to make it more clear.